### PR TITLE
Publish JDT warnings to problems view

### PR DESCRIPTION
**Pull request by simontoens**
Note that there are 2 commits: one of them only removes unused imports.

### CODE REVIEW

File: plugin-core/plugin.xml [hunk 1]
```diff
@@ -73,7 +73,7 @@
    <extension id="com.salesforce.bazel.eclipse.bazelmarker"
          point="org.eclipse.core.resources.markers">
          <super type="org.eclipse.core.resources.problemmarker"/>
-         <persistent value="true"/>
+         <persistent value="false"/>
```
Comment (simontoens) [inline plugin-core/plugin.xml:76]: We don't need to persist problem makers - they are recomputed when building.

File: plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelBuilder.java [hunk 2]
```diff
@@ -86,20 +88,28 @@
 
     public static final String BUILDER_NAME = "com.salesforce.bazel.eclipse.builder";
 
+    private static final AtomicBoolean REGISTERED_EL_CHANGE_LISTENER = new AtomicBoolean(false);
     private static final LogHelper LOG = LogHelper.log(BazelBuilder.class);
 
+    public BazelBuilder() {
+        if (!REGISTERED_EL_CHANGE_LISTENER.getAndSet(true)) {
+            // is there a better way/place to register a singleton?
+            JavaCore.addElementChangedListener(new JDTWarningPublisher());
```
Comment (simontoens) [inline plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelBuilder.java:97]: @plaird let me know if you know of a real hook I can put this in.  I just want a single instance of JDTWarningPublisher registered.  I thought BazelNature may work, but nothing in there is called when Eclipse just restarts, with a bazel project already imported.  BazelBuilder is create for each project, that's why I added the boolean gate.

Comment (plaird) [reply plugin-core/src/main/java/com/salesforce/bazel/eclipse/builder/BazelBuilder.java:97]: BazelPluginActivator.startInternal is another candidate location.

**Comment by plaird** [comment]
Tested this locally, it works great for files open in the editor. It doesn't list JDT warnings in files not opened by the editor. I think that is ok.

**Comment by simontoens** [comment]
@plaird thanks.  Yes it only works for files open in the editor ... I think we can always revisit and try to improve.

**Pull request merged**